### PR TITLE
[PM-3177] Extract IOrganizationService.UpdateUserGroupsAsync to a command

### PR DESCRIPTION
**Pull request by r-tome**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Extract `OrganizationService.UpdateUserGroupsAsync` to a command.

## Code changes

* **bitwarden_license/src/Scim/Utilities/ScimServiceCollectionExtensions.cs:** `IDeleteOrganizationUserCommand` DI registration moved to `OrganizationServiceCollectionExtensions`
* **src/Api/Controllers/OrganizationUsersController.cs:** Replaced `IOrganizationService.UpdateUserGroupsAsync` with `IUpdateOrganizationUserGroupsCommand`
* **src/Api/Public/Controllers/MembersController.cs:** Replaced `IOrganizationService.UpdateUserGroupsAsync` with `IUpdateOrganizationUserGroupsCommand`
* **src/Core/OrganizationFeatures/OrganizationServiceCollectionExtensions.cs:** Registering commands for DI
* **src/Core/OrganizationFeatures/OrganizationUsers/Interfaces/IUpdateOrganizationUserGroupsCommand.cs:** Command interface
* **src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs:** Abstract class that contains methods to be shared amongst all `OrganizationUser` commands
* **src/Core/OrganizationFeatures/OrganizationUsers/UpdateOrganizationUserGroupsCommand.cs:** Command with code extracted from `OrganizationService`
* **src/Core/Services/IOrganizationService.cs:** Deleted method `UpdateUserGroupsAsync` that was extracted
* **src/Core/Services/Implementations/OrganizationService.cs:** Deleted method `UpdateUserGroupsAsync` that was extracted
* **test/Core.Test/OrganizationFeatures/OrganizationUsers/UpdateOrganizationUserGroupsCommandTests.cs:** Added unit tests for `UpdateOrganizationUserGroupsCommand`

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs
Comment (eliykat) [inline src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs:1]: I assume these methods are temporarily duplicated from `OrganizationService`, and in your final PR you'll remove them from the service?

I prefer to do this the other way around - call the existing `OrganizationService` methods from the command/s, then in the final PR bring it into the base class (or wherever it's required). This avoids unnecessary duplication in the meantime, and when we move it'll be easier to confirm that's where it belongs.

Incidentally, just about everything here looks like it's checking permissions, so it should go into an `AuthorizationHandler`. We don't need to do that here (to limit scope & get some of this merged), just mentioning it.

**Review by eliykat — CHANGES_REQUESTED** [review]

**Comment by eliykat** [comment]
Looks good, just 1 comment above.

File: src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs
Comment (r-tome) [reply src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs:1]: > I assume these methods are temporarily duplicated from `OrganizationService`, and in your final PR you'll remove them from the service?

Yes, that was the idea.

>I prefer to do this the other way around - call the existing `OrganizationService` methods from the command/s, then in the final PR bring the methods into the base class (or wherever they're required).

I like that! I'll change it.

Comment (r-tome) [reply src/Core/OrganizationFeatures/OrganizationUsers/OrganizationUserCommand.cs:1]: I have made this change and also added a few unit tests.

File: test/Core.Test/Services/OrganizationServiceTests.cs
Comment (eliykat) [inline test/Core.Test/Services/OrganizationServiceTests.cs:1]: I love simple tests 😁

**Review by eliykat — APPROVED** [review]

**Pull request merged**